pkg/sql/executor_ir.go

```diff
@@ -61,11 +61,8 @@ func RunSQLProgram(sqlProgram string, modelDir string, session *pb.Session) *pip
 		}
 		err = runSQLProgram(wr, sqlProgram, db, modelDir, session)
 		if err != nil {
-			err = wr.Write(fmt.Errorf("runSQLProgram error: %v", err))
-			if err != pipe.ErrClosedPipe {
-				if err := wr.Write(err); err != nil {
```

Comment:
The root reason is written `nil` into the pipe at L66, and step panic.

Reply:
My apology for introducing this on https://github.com/sql-machine-learning/sqlflow/pull/1983/files#diff-45b52a537a003bab57099e84c65c0985R64